tools/serverpod_cli/test/analyzer/entities/entity_analyzer/relation_keyword_test.dart

```diff
@@ -5,6 +5,160 @@ import 'package:serverpod_cli/src/util/protocol_helper.dart';
 import 'package:test/test.dart';
 
 void main() {
+  group('Given a class with a self relation on a field with the class datatype',
+      () {
+    var collector = CodeGenerationCollector();
+
+    var protocol = ProtocolSource(
+      '''
+class: Example
+table: example
+fields:
+  parent: Example?, relation
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol,
+    );
+    SerializableEntityAnalyzer.resolveEntityDependencies([definition!]);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol.yaml,
+      protocol.yamlSourceUri.path,
+      collector,
+      definition,
+      [definition],
+    );
+
+    var classDefinition = definition as ClassDefinition;
+
+    test('then no errors are collected.', () {
+      expect(collector.errors, isEmpty);
+    });
+    test('then the table is not set on relation field.', () {
+      var parent = classDefinition.findField('parent');
+      expect(parent?.parentTable, null);
```

Comment:
Question - For my knowledge, `parentTable` is the bool that defines if we store it in the database?

Reply:
No `parentTable` is the name of the parent table, I.E. it is a string.